extensions/amp-story-shopping/0.1/test/test-amp-story-shopping-attachment.js

```diff
@@ -26,113 +28,205 @@ describes.realWin(
-        'amp-story-page-attachment'
-      );
-      const attachmentChildImpl = await attachmentChildEl.getImpl();
-      env.sandbox.stub(attachmentChildImpl, 'mutateElement').callsFake(() => {
-        expect(pageEl.querySelector('.amp-story-shopping-plp')).to.not.be.null;
-      });
+    it('should open attachment when active product data is set', async () => {
+      await shoppingImpl.layoutCallback();
+      await attachmentChildImpl.layoutCallback();
+      dispatchActiveProductData();
+      // Simulating the getImpl in amp-story-shopping-attachment's
+      // onShoppingDataUpdate_ method
+      await attachmentChildEl.getImpl();
+      expect(
+        attachmentChildEl.classList.contains(
+          'i-amphtml-story-draggable-drawer-open'
+        )
+      ).to.be.true;
     });
 
-    it('should build PLP with data from tag on page', async () => {
-      await dispatchTestShoppingData();
-      const attachmentChildEl = shoppingEl.querySelector(
-        'amp-story-page-attachment'
+    it('should build PDP if active product data', async () => {
+      await shoppingImpl.layoutCallback();
+      await attachmentChildImpl.layoutCallback();
+      dispatchActiveProductData();
+      // Simulating the getImpl in amp-story-shopping-attachment's
+      // onShoppingDataUpdate_ method
+      await attachmentChildEl.getImpl();
+      expect(
+        attachmentChildEl.querySelector('.i-amphtml-amp-story-shopping-pdp')
+      ).to.not.be.null;
+    });
+
+    it('should build PDP on PLP card click', async () => {
+      await shoppingImpl.layoutCallback();
+      await attachmentChildImpl.layoutCallback();
+      storeService.dispatch(Action.TOGGLE_PAGE_ATTACHMENT_STATE, true);
+      const plpCard = attachmentChildEl.querySelector(
+        '.i-amphtml-amp-story-shopping-plp-card'
       );
-      const attachmentChildImpl = await attachmentChildEl.getImpl();
-      env.sandbox.stub(attachmentChildImpl, 'mutateElement').callsFake(() => {
-        expect(
-          pageEl.querySelector(
-            '.amp-story-shopping-plp-card .amp-story-shopping-plp-card-title'
-          ).textContent
-        ).to.equal('Spectacular Spectacles');
+      plpCard.dispatchEvent(new Event('click'));
+      expect(
+        attachmentChildEl.querySelector('.i-amphtml-amp-story-shopping-pdp')
+      ).to.not.be.null;
+    });
+
+    it('should default to PDP on open if only one product for the page', async () => {
+      await shoppingImpl.layoutCallback();
+      await attachmentChildImpl.layoutCallback();
+      // Override to simulate data for one product on the page.
+      storeService.dispatch(Action.ADD_SHOPPING_DATA, {
+        page1: {
+          [shoppingData.productId]: shoppingData.items[0],
+        },
       });
+      storeService.dispatch(Action.TOGGLE_PAGE_ATTACHMENT_STATE, true);
+      expect(
+        attachmentChildEl.querySelector('.i-amphtml-amp-story-shopping-pdp')
+      ).to.not.be.null;
```

Comment:
Here and above, have you considered what's best in expecting the selector to target something, vs looking at the `activeProductData`?

Reply:
We could trigger re-opening the attachment to ensure it opens from a PLP, but that is covered in a test already 🤔
I like how this test is specific to when the activeProductData is cleared.